crates/bitwarden/src/crypto/asymmetric_crypto_key.rs

```diff
@@ -0,0 +1,99 @@
+use std::str::FromStr;
+
+use base64::Engine;
+use rsa::RsaPrivateKey;
+
+use crate::{
+    crypto::CryptoKey,
+    error::{CryptoError, Error, Result},
+    util::BASE64_ENGINE,
+};
+
+/// An asymmetric encryption key. Used to encrypt and decrypt [`EncString`](crate::crypto::EncString)
+pub struct AsymmetricCryptoKey {
+    pub(in crate::crypto) key: RsaPrivateKey,
+}
+
+impl AsymmetricCryptoKey {
+    pub fn from_pem(pem: &str) -> Result<Self> {
+        use rsa::pkcs8::DecodePrivateKey;
+        Ok(Self {
+            key: rsa::RsaPrivateKey::from_pkcs8_pem(pem).map_err(|_| CryptoError::InvalidKey)?,
+        })
+    }
+}
+
+impl FromStr for AsymmetricCryptoKey {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let bytes = BASE64_ENGINE
+            .decode(s)
+            .map_err(|_| CryptoError::InvalidKey)?;
+        AsymmetricCryptoKey::try_from(bytes.as_slice())
+    }
+}
+
+impl TryFrom<&[u8]> for AsymmetricCryptoKey {
+    type Error = Error;
+
+    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
+        use rsa::pkcs8::DecodePrivateKey;
+        Ok(Self {
+            key: rsa::RsaPrivateKey::from_pkcs8_der(value).map_err(|_| CryptoError::InvalidKey)?,
+        })
+    }
+}
```

Comment:
Initially I had a FromStr for the pem format as well, but then was worried it might be confused for the FromStr on SymmetricCryptoKey which is just base64 encoded bytes, so decided to go with an explicit alternative.

It's probably a good idea to be explicit in both cases though, yeah.